lib/qx/tool/compiler/utils/Json.js

```diff
@@ -41,15 +43,68 @@ qx.Class.define("qx.tool.compiler.utils.Json", {
       if (!str) {
         return null;
       }
+      // eslint-disable-next-line no-path-concat
+      var qxRoot = __dirname + "/../../../../../";
       var ast = JsonToAst.parseToAst(str);
       var json = JsonToAst.astToObject(ast);
+      var id = this.identifyData(json);
+      if (id) {
+        var ajv = new Ajv({
+          allErrors: true,
+          jsonPointers: true
+        });
+        var schema = "https://raw.githubusercontent.com/qooxdoo/qooxdoo-compiler/master/resource/" + id.type + "-" + String(id.version) + ".json";
+        var valid = ajv
+          .addSchema(require(qxRoot + "resource/compile-" + String(id.version) + ".json"))
+          .addSchema(require(qxRoot + "resource/Manifest-" + String(id.version) + ".json"))
```

Comment:
`String(...)` is not required, as numbers are converted to strings when concatenated to other strings.